src/compatmacro.jl

```diff
@@ -116,3 +116,38 @@ function _destructure_named_tuple(ex::Expr)
     push!(ex.args, values)
     return ex
 end
+
+# https://github.com/JuliaLang/julia/pull/50105
+macro compat(public::Symbol, symbols_expr::Union{Expr, Symbol})
+    public == :public || throw(ArgumentError("Invalid Syntax: `@compat $public $symbols_expr`"))
+    symbols = _get_symbols(symbols_expr)
+    if VERSION >= v"1.11.0-DEV.469"
```

Comment:
```suggestion
    if VERSION >= v"1.11.0-DEV.469"
        symbols = _get_symbols(symbols_expr)
```
This should fix it, as the old julia versions no longer traverse the faulty path via `_get_symbols`.

Reply:
I want to keep the parsing outside the version check so that folks get errors consistently on all versions.
With that change, `@compat 1+1 == 2` would expand to `nothing` on old versions of Julia and throw on new versions of Julia.

I've tried to implement a fix that actually supports `@var_str` while still throwing on invalid inputs on all versions.

Reply:
Great! That sounds like a good idea :)